methods.py

```diff
@@ -81,6 +80,34 @@ def add_source_files(self, sources, files, allow_gen=False):
     return True
 
 
+def redirect_emitter(target, source, env):
+    """
+    Emitter to automatically redirect object/library build files to the `bin/obj` directory,
+    retaining subfolder structure. External build files will attempt to retain subfolder
+    structure relative to their environment's parent directory, sorted under `bin/obj/external`.
+    If `redirect_build_objects` is `False`, or an external build file isn't relative to the
+    passed environment, this emitter does nothing.
+    """
+    if not env["redirect_build_objects"]:
+        return target, source
+    redirected_targets = []
+    for item in env.Flatten(target):
+        if isinstance(item, str):
+            item = env.File(item)
```

Comment:
I think it's worth a comment to explain that `string` targets are always paths, and `SCons` would have converted them to `File` later anyway.